oid4vci/integration/credo/index.ts

```diff
@@ -0,0 +1,168 @@
+        registrars: [new PeerDidRegistrar()],
+        resolvers: [new PeerDidResolver()]
+      }),
+      openId4VcHolderModule: new OpenId4VcHolderModule(),
+      w3cCredentials: new W3cCredentialsModule(),
+    },
+  });
+
+  await agent.initialize();
+  return {};
+});
+
+
+const getAgent = () => {
+  if (agent === null) {
+    throw new Error('Agent not initialized');
+  }
+  return agent;
+};
+
+proxy.rpc.addMethod(
+  'openid4vci.acceptCredentialOffer',
+  async ({offer}: {offer: string}) => {
+    const agent = getAgent();
+
+    // resolved credential offer contains the offer, metadata, etc..
+    const resolvedCredentialOffer = await agent.modules.openId4VcHolderModule.resolveCredentialOffer(offer)
+    console.log('Resolved credential offer', JSON.stringify(resolvedCredentialOffer.credentialOfferPayload, null, 2))
+
+    // issuer only supports pre-authorized flow for now
+    const credentials = await agent.modules.openId4VcHolderModule.acceptCredentialOfferUsingPreAuthorizedCode(
+      resolvedCredentialOffer,
+      {
+        credentialBindingResolver: async ({
+          supportedDidMethods,
+          keyType,
+          supportsAllDidMethods,
+          // supportsJwk now also passed
+          supportsJwk,
+          credentialFormat,
+        }: {
+          supportedDidMethods: any,
+          keyType: any,
+          supportsAllDidMethods: any,
+          // supportsJwk now also passed
+          supportsJwk: any,
+          credentialFormat: any,
+        }) => {
+          // NOTE: example implementation. Adjust based on your needs
+          // Return the binding to the credential that should be used. Either did or jwk is supported
+
+          if (supportsAllDidMethods || supportedDidMethods?.includes('did:key')) {
+            const didResult = await agent.dids.create<KeyDidCreateOptions>({
+              method: 'key',
+              options: {
+                keyType,
+              },
+            })
+
+            if (didResult.didState.state !== 'finished') {
+              throw new Error('DID creation failed.')
+            }
+
+            const didKey = DidKey.fromDid(didResult.didState.did)
+
+            return {
+              method: 'did',
+              didUrl: `${didKey.did}#${didKey.key.fingerprint}`,
+            }
+          }
+
+          // we also support plain jwk for sd-jwt only
+          if (supportsJwk && credentialFormat === OpenId4VciCredentialFormatProfile.SdJwtVc) {
+            const key = await agent.wallet.createKey({
+              keyType,
+            })
+
+            // you now need to return an object instead of VerificationMethod instance
+            // and method 'did' or 'jwk'
+            return {
+              method: 'jwk',
+              jwk: getJwkFromKey(key),
+            }
+          }
+
+          throw new Error('Unable to create a key binding')
+        },
+      }
+    )
+
+    console.log('Received credentials', JSON.stringify(credentials, null, 2))
```

Comment:
Is this intentional?

Reply:
Yeah, handy for debugging; since this is meant purely for testing, I think it's valuable to keep around.